dlt/common/configuration/inject.py

```diff
@@ -58,6 +65,7 @@ def with_config(
     include_defaults: bool = True,
     accept_partial: bool = False,
     initial_config: Optional[BaseConfiguration] = None,
```

Comment:
should we have a "thread_safe" parameter so we only lock on decorated functions where we know that there might be issues?